Sources/Citadel/TTY/Client/TTY.swift

```diff
@@ -96,6 +96,75 @@ public enum ExecCommandOutput {
     case stderr(ByteBuffer)
 }
 
+final class ShellCommandHandler: ChannelDuplexHandler {
+    enum Output {
+        case shellSuccess
+        case output(ByteBuffer)
+        case eof(Error?)
+    }
+    
+    typealias InboundIn = SSHChannelData
+    typealias InboundOut = ByteBuffer
+    typealias OutboundIn = ByteBuffer
+    typealias OutboundOut = SSHChannelData
+
+    let logger: Logger
+    let onOutput: (Channel, Output) -> ()
+    
+    init(logger: Logger, onOutput: @escaping (Channel, Output) -> ()) {
+        self.logger = logger
+        self.onOutput = onOutput
+    }
+    
+    func handlerAdded(context: ChannelHandlerContext) {
+        context.channel.setOption(ChannelOptions.allowRemoteHalfClosure, value: true).whenFailure { error in
+            context.fireErrorCaught(error)
+        }
+    }
+
+    func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
+        switch event {
+        case is SSHChannelRequestEvent.ExitStatus:
+            ()
+        case is NIOSSH.ChannelSuccessEvent:
+            onOutput(context.channel, .shellSuccess)
+        case is NIOSSH.ChannelFailureEvent:
+            onOutput(context.channel, .eof(CitadelError.shellRequestFailed))
+            context.close(promise: nil)
+        default:
+            context.fireUserInboundEventTriggered(event)
+        }
+    }
+    
+    func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+        let data = self.unwrapInboundIn(data)
+        
+        guard case .byteBuffer(let buffer) = data.data else {
+            logger.error("Unable to process channelData for executed command. Data was not a ByteBuffer")
+            return onOutput(context.channel, .eof(SSHExecError.invalidData))
+        }
+        
+        switch data.type {
+        case .channel:
+            onOutput(context.channel, .output(buffer))
+        case .stdErr:
+            onOutput(context.channel, .output(buffer))
+        default:
+            // We don't know this std channel
```

Comment:
Should we consider throwing an error here?

Reply:
I copied this from the `ExecCommandHandler`. I'm not familiar enough with SSH to know if there could be non-error channel data coming in that is safe to ignore and shouldn't stop the whole thing. In any case if we want to adjust this we should apply it to the other handlers as well. WDYT?